src/knex-store.ts

```diff
@@ -0,0 +1,251 @@
+/* Copyright (c) 2010-2022 Richard Rodger and other contributors, MIT License */
+'use strict'
+
+// TODO: use `undefined` as no-error value consistently
+
+import { intern } from './intern'
+
+const knex = require('knex')({
+  client: 'pg',
+  version: '8.8.0'
+})
+
+let internals = {
+  name: 'knex-store',
+}
+
+module.exports = knex_store
+Object.defineProperty(module.exports, 'name', { value: 'knex-store' })
+
+module.exports.defaults = {
+  'entity-id-exists':
+    'Entity of type <%=type%> with id = <%=id%> already exists.',
+}
+
+/* NOTE: `intern` serves as a namespace for utility functions used by
+ * the knex store.
+ */
+module.exports.intern = intern
+
+type Options = {
+  prefix: string
+  idlen: number
+  web: {
+    dump: boolean
+  }
+  generate_id: any
+}
+
+function knex_store(this: any, options: Options) {
+  
+  // Take a reference to the calling Seneca instance
+  let seneca: any = this
+  
+  let conf = intern.getConfig(seneca)
+  knex.conection = conf
+
+  let init = seneca.export('entity/init')
+
+  // merge default options with any provided by the caller
+  options = seneca.util.deepextend(
+    {
+      prefix: '/knex-store',
+      idlen: 6,
+      web: {
+        dump: false,
+      },
+
+      // TODO: use seneca.export once it allows for null values
+      generate_id: seneca.root.private$.exports['entity/generate_id'],
+    },
+    options
+  )
+
+  // The calling Seneca instance will provide
+  // a description for us on init(), it will
+  // be used in the logs
+  let desc: any
+
+  // Our super awesome in knex database. Please bear in mind
+  // that this store is meant for fast prototyping, using
+  // it for production is not advised!
+  let entmap: any = {}
```

Comment:
I kept a few lines of mem-store code that I didn't quite understand what it was for, removed

Reply:
mem-store is just an in-memory simulation of a database using JS objects - not needed when we have a real database - review the code of https://github.com/senecajs/seneca-postgres-store for example

Reply:
This I understood, I said in relation to this line of code, I didn't know that it belonged to the mem-store because I didn't know the purpose of `entmap` variable